examples/example-gateway/middlewares/example/example.go

```diff
@@ -46,9 +50,13 @@ type MiddlewareState struct {
 // NewMiddleWare creates a new middleware that executes the next middleware
 // after performing it's operations.
 func NewMiddleWare(
```

Comment:
let's not change this API and keep it as `*zanzibar.Gateway`

Reply:
Sure.

Reply:
Sure, but the middleware logger and scope won't have the endpoint fields and tags.